### PR TITLE
Handle validation message

### PR DESCRIPTION
**Pull request by slevertbiot**
Added logic to get ready to handle validation messages.
If we can't handle a message, because it's a type we are not ready to handle, we will log it.
Refactored tests to make them simpler.

### CODE REVIEW

File: udmif/event-handler/src/UdmiMessageHandler.ts [hunk 1]
```diff
@@ -3,26 +3,42 @@ import { DeviceDocument } from './DeviceDocument';
 import { createDeviceDocument } from './DeviceDocumentFactory';
 import { UdmiMessage } from './UdmiMessage';
 import { DeviceKey } from './DeviceKey';
+import { isPointsetSubType, isSystemSubType, isValidationSubType } from './MessageUtils';
 
 export default class UdmiMessageHandler {
   constructor(private deviceDao: DeviceDao) { }
 
   async handleUdmiEvent(udmiMessage: UdmiMessage) {
```
Comment (mspratt-biot) [inline udmif/event-handler/src/UdmiMessageHandler.ts:11]: Shouldn't this have a return type void?

File: udmif/event-handler/src/UdmiMessageHandler.ts [hunk 2]
```diff
@@ -3,26 +3,42 @@ import { DeviceDocument } from './DeviceDocument';
...
+        await this.handleMessage(udmiMessage);
+      } else {
+        console.warn('Could not handle the following message: ' + JSON.stringify(udmiMessage));
+      }
+    } catch (e) {
+      console.error('An unexpected error occurred: ', e);
+    }
+  }
+
+  private async handleMessage(udmiMessage: UdmiMessage) {
```
Comment (mspratt-biot) [inline udmif/event-handler/src/UdmiMessageHandler.ts:23]: Return type void?

File: udmif/event-handler/src/UdmiMessageHandler.ts [hunk 3]
```diff
@@ -3,26 +3,42 @@ import { DeviceDocument } from './DeviceDocument';
...
       const document: DeviceDocument = createDeviceDocument(udmiMessage, existingDeviceDocument?.points || []);
       this.deviceDao.upsert(deviceKey, document);
-    } catch (e) {
-      console.error('An unexpected error occurred: ', e);
     }
   }
 }
 
 export function getDeviceKey(message: UdmiMessage): DeviceKey {
+
```
Comment (mspratt-biot) [inline udmif/event-handler/src/UdmiMessageHandler.ts:34]: Remove blank space.

File: udmif/event-handler/src/UdmiMessageHandler.ts [hunk 4]
```diff
@@ -3,26 +3,42 @@ import { DeviceDocument } from './DeviceDocument';
...
   if (!message.attributes.deviceId || !message.attributes.deviceNumId) {
     throw new Error('An invalid device name or id was submitted');
   }
 
   return { name: message.attributes.deviceId, id: message.attributes.deviceNumId };
 }
+
+export function messageCanBeHandled(message: UdmiMessage): boolean {
```
Comment (mspratt-biot) [inline udmif/event-handler/src/UdmiMessageHandler.ts:42]: Will this be used outside of this class? If not, move it into the class as private. Same goes for getDeviceKey.

File: udmif/event-handler/src/UdmiMessageHandler.ts (hunk 1 above)
Comment (slevertbiot) [reply udmif/event-handler/src/UdmiMessageHandler.ts:11]: I can have a return type of void to make it more explict.

Comment (mspratt-biot) [reply udmif/event-handler/src/UdmiMessageHandler.ts]: Just used to the angular project where this is enforced...

**Review by mspratt-biot — APPROVED** [review]

**Pull request merged**